src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/DeclarationImpl.java

```diff
@@ -176,6 +191,20 @@ public Function withAttributes(Map<String, List<Constable>> attrs) {
         public Function stripAttributes() {
             return new FunctionImpl(type, params, name(), pos(), null);
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            FunctionImpl function = (FunctionImpl) o;
+            return params.equals(function.params) &&
```

Comment:
Type would have covered both parameter type and return type. Compare params depends on how Variable implements equals and may cause same type declaration with different argument name to fail(not the case here).

Reply:
Agree. Function parameters should not be compared. Only type and (function) name should be. Will fix this and expand test to cover this case (void func(int x) and void func(int y) as duplicate declarations)